app/controllers/api/base_controller/renderer.rb

```diff
@@ -514,7 +514,13 @@ def api_user_role_allows?(action_identifier)
 
         return custom_api_user_role_allows?(action_identifier) if custom_api_user_role_allows_method?(action_identifier)
 
-        Array(action_identifier).any? { |identifier| User.current_user.role_allows?(:identifier => identifier) }
+        @role_allows_cache ||= {}
+        Array(action_identifier).any? do |identifier|
+          if @role_allows_cache.key?(identifier)
```

Comment:
Is the `if` reversed?  Am I missing something?  How will we ever set the `@role_allows_cache` on line 520?

Reply:
Yes... yes it is.  I didn't even copy what I had above.

Welp, we know the specs fail as a result.  Fixing...